test/clientconn_state_transition_test.go

```diff
@@ -616,3 +618,53 @@ func (s) TestConnectivityStateSubscriber(t *testing.T) {
 		}
 	}
 }
+
+// TestChannelStateWaitingForFirstResolverUpdate verifies the initial
+// state of the channel when a manual name resolver doesn't provide any updates.
+func (s) TestChannelStateWaitingForFirstResolverUpdate(t *testing.T) {
+	backend := stubserver.StartTestService(t, nil)
+	defer backend.Stop()
+
+	mr := manual.NewBuilderWithScheme("e2e-test")
+	defer mr.Close()
+
+	cc, err := grpc.NewClient(mr.Scheme()+":///", grpc.WithResolvers(mr), grpc.WithTransportCredentials(insecure.NewCredentials()))
+	if err != nil {
+		t.Fatalf("Failed to create new client: %v", err)
+	}
+	defer cc.Close()
+
+	ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
+	defer cancel()
+
+	shortCtx, shortCancel := context.WithTimeout(ctx, defaultTestShortTimeout)
+	defer shortCancel()
+
+	// TODO: Change this assertion to testutils.AwaitState(shortCtx, t, cc,
+	// connectivity.Connecting) when the channel correctly transitions to
+	// CONNECTING while waiting for the first resolver update.
+	testutils.AwaitNoStateChange(shortCtx, t, cc, connectivity.Idle)
+
+	internal.EnterIdleModeForTesting.(func(*grpc.ClientConn))(cc)
+
+	cc.Connect()
+
+	testutils.AwaitNoStateChange(shortCtx, t, cc, connectivity.Idle)
+
+	shortCancel()
+	shortCtx, shortCancel = context.WithTimeout(ctx, defaultTestShortTimeout)
+	defer shortCancel()
+
+	go func() {
+		_, err := testgrpc.NewTestServiceClient(cc).EmptyCall(shortCtx, &testgrpc.Empty{})
+		if err == nil {
+			t.Errorf("Expected RPC to fail, but it succeeded")
+		}
+	}()
+
+	shortCancel()
+	shortCtx, shortCancel = context.WithTimeout(ctx, defaultTestShortTimeout)
+	defer shortCancel()
```

Comment:
You can remove the re-creation of the context here and keep using the same context. `testutils.AwaitNoStateChange` will wait for `defaultTestShortTimeout` while the go routine is making an rpc in the background. After the timeout, the test function will return and the context will be cancelled due to the deferred statement. This will cause the rpc in the go routine to also fail and return.